### ISSUE TITLE
ANCM: the global aspnetcorerh.dll should not be used for inprocess mode even though after switching process model from out-of-process mode

### ISSUE THREAD

**Issue by jhkimnew** [aspnet/IISIntegration#790]
STEPS TO REPRO:

0.	Make sure aspnetcorerh.dll is *not* placed on SDK folder (nor application folder) first so that only the global aspnetcorerh.dll exists
1.	Copy aspnetcore.dll and aspnetcorerh.dll of V2 on %windir%\system32\inetsrv
2.	Send a request to an aspnetcore app which is running in out-of-process mode
3.	Switch the hosting model to inprocess for the aspnetcore app
4.	Send the same request

EXPETED:

500 error and there should be an even error that aspnetcorerh.dll not found.

ACTUAL:

No error. 200 retrieved.

**Comment by jhkimnew** [comment]
I found another issue which seems to be fixed together when we fix this issue. So, when this issue is fixed, we need to verify the new issue is fixed also.

The repro step of the new issue.
1. Send a request 
2. Switch the process mode from inprocess to outofprocess
3. Send a request 
4. Switch the process mode from outofprocess to inprocess

EXPECTED:
There is no 500 error

ACTUAL:
500 error happens and I got some event error as the following:

<><><>
Application '/LM/W3SVC/1/ROOT/aspnetcoreapprc' with physical root 'D:\inetpub\wwwroot\aspnetcoreapprc\' hit unexpected managed background thread exit, ErrorCode = '0x8027025b. Last 4KB characters of captured stdout and stderr logs:
<><><>
